widget/radio_test.go

```diff
@@ -343,6 +344,39 @@ func TestRadio_FocusIndicator_Centered_Vertically(t *testing.T) {
 	}
 }
 
+func TestRadio_Mandatory(t *testing.T) {
+	radio := NewRadio([]string{"Hi", "There"}, func(string) {})
+	radio.SetMandatory(true)
+
+	assert.True(t, radio.Mandatory)
+	assert.Equal(t, "Hi", radio.Selected, "mandatory radios select the first option if none is selected")
+
+	radio = NewRadio([]string{"Hi", "There"}, func(string) {})
+	radio.SetSelected("There")
+	radio.SetMandatory(true)
+
+	assert.True(t, radio.Mandatory)
+	assert.Equal(t, "There", radio.Selected, "radio becoming mandatory does not affect a valid selection")
+
+	radio.SetSelected("")
+	assert.True(t, radio.Mandatory)
+	assert.Equal(t, "Hi", radio.Selected, "unsetting selection of mandatory radio selects the first option")
```

Comment:
This doesn't feel right. If I have previously made a selection and something triggers a reset then it should either clear, or remain the same. Setting it to something different feels unexpected. So I think this should check for "There".

Reply:
With the removal of `SetMandatory` I will also remove the adjustment of the selection if it is set to `""`.